ephios/static/plugins/basesignup/js/disposition.js

```diff
@@ -0,0 +1,99 @@
+        $form.find("[data-show-for-state]").each((index, el) => {
+            el = $(el);
+            if (el.attr("data-show-for-state").split(",").includes(state.toString())) {
+                el.slideDown();
+            } else if (!instant) {
+                el.slideUp();
+            } else {
+                el.fadeOut(0);
+            }
+        });
+    }
+
+    $("[data-drop-to-state]").each(function (index, elem) {
+        const newState = $(elem).data("drop-to-state");
+        Sortable.create(elem, {
+            group: "participations",
+            sort: true,
+            draggable: ".draggable",
+            emptyInsertThreshold: 20,
+            fallbackTolerance: 5,
+            animation: 150,
+            easing: "cubic-bezier(1, 0, 0, 1)",
+            // scrolling
+            scrollSensitivity: 150,
+            scrollSpeed: 15,
+            // set state
+            onAdd: (event) => {
+                $(event.item).find(".state-input").val(newState);
+                handleDispositionForm($(event.item), newState, false);
+            },
+        });
+        handleDispositionForm($(elem), newState, true);
+    });
+
+    $("select#id_user[form='add-user-form']").on('select2:close', function () {
+        // clear select2
+        const userSelect = $(this);
+        setTimeout(() => {
+            userSelect.val(null).change()
+        });
+
+        const spawn = $("[data-formset-spawn]");
+        const formset = $('#participations-form').formset('getOrCreate');
+        // look for existing form with that participation
+        const userId = $(this).val();
+        if (!userId) {
+            return;
+        }
+
+        const participation = $('[data-participant-id=' + userId + ']');
+        if (participation.length) {
+            // we already have that card
+            const prefix = formset.extractPrefix(participation);
+            const deleteCheckbox = participation.find('[name=' + prefix + '-DELETE]');
+            if (deleteCheckbox.attr("checked")) {
+                // was marked for deletion, so revert that.
+                participation.attr("data-formset-created-at-runtime", true); // so formset.js decides to slideDown() it
+                deleteCheckbox.attr("checked", false).change();
+            }
+            // now visible. Move it to here.
+            $([document.documentElement, document.body]).animate({
+                scrollTop: participation.offset().top - 200
+            }, 1000);
+            participation.focus();
+            participation.addClass("list-group-item-info");
+            setTimeout(() => {
+                participation.removeClass("list-group-item-info")
+            }, 2000);
+        } else {
+            // get the new form from the server
+            const addUserForm = $("#" + $(this)[0].form.id);
+            const newIndex = formset.totalFormCount();
+            addUserForm.find('[name=new_index]').val(newIndex);
+            $.ajax({
```

Comment:
Do we need a loading indicator/disable the input while the request is running?

Reply:
i think just disabling wouldn't be a good UX. some sort of spinner would be cool, but I can't think of a nice way to place it. If you also can't come up with one now I'd also be fine with moving that to another issue.

Reply:
for me it looks like the added participation is displayed and hidden afterwards, that looks kinda strange.